### PR TITLE
fix: reschedule pings problem

### LINKED ISSUE #1903: [Bug]: Keepalive timeout error always occurs when reschedulePings == false.
### MQTTjs Version

5.8.0

### Broker

mosquitto

### Environment

NodeJS

### Description

When I configure reschedulePings == false, keepalive timeout error always occurs though pingresp received.
If the client receives pingresp message, it should reset the KeepaliveManager timer regardless of reschedulePings setting.

### Minimal Reproduction

```js
const mqtt = require('mqtt');

const client = mqtt.connect('mqtt://localhost:1883', 
    {
        reschedulePings: false,
        keepalive: 1
    }
);

client.on('error', (err) => {
    console.log(err);
    client.end()
});
```

### Debug logs

```
> node keepalive_test.js
  mqttjs connecting to an MQTT broker... +0ms
  mqttjs:client MqttClient :: version: 5.8.0 +0ms
  mqttjs:client MqttClient :: environment node +0ms
  mqttjs:client MqttClient :: options.protocol mqtt +0ms
  mqttjs:client MqttClient :: options.protocolVersion 4 +0ms
  mqttjs:client MqttClient :: options.username undefined +1ms
  mqttjs:client MqttClient :: options.keepalive 1 +0ms
  mqttjs:client MqttClient :: options.reconnectPeriod 1000 +0ms
  mqttjs:client MqttClient :: options.rejectUnauthorized undefined +0ms
  mqttjs:client MqttClient :: options.properties.topicAliasMaximum undefined +0ms
  mqttjs:client MqttClient :: clientId mqttjs_4713df34 +0ms
  mqttjs:client MqttClient :: setting up stream +0ms
  mqttjs:client connect :: calling method to clear reconnect +1ms
  mqttjs:client _clearReconnect : clearing reconnect timer +0ms
  mqttjs:client connect :: using streamBuilder provided to client to create stream +0ms
  mqttjs calling streambuilder for mqtt +4ms
  mqttjs:tcp port 1883 and host localhost +0ms
  mqttjs:client connect :: pipe stream to writable stream +11ms
  mqttjs:client connect: sending packet `connect` +1ms
  mqttjs:client _writePacket :: packet: {
  mqttjs:client   cmd: 'connect',
  mqttjs:client   protocolId: 'MQTT',
  mqttjs:client   protocolVersion: 4,
  mqttjs:client   clean: true,
  mqttjs:client   clientId: 'mqttjs_4713df34',
  mqttjs:client   keepalive: 1,
  mqttjs:client   username: undefined,
  mqttjs:client   password: undefined,
  mqttjs:client   properties: undefined
  mqttjs:client } +0ms
  mqttjs:client _writePacket :: emitting `packetsend` +1ms
  mqttjs:client _writePacket :: writing to stream +0ms
  mqttjs:client _writePacket :: writeToStream result true +12ms
  mqttjs:client writable stream :: parsing buffer +4ms
  mqttjs:client parser :: on packet push to packets array. +0ms
  mqttjs:client work :: getting next packet in queue +0ms
  mqttjs:client work :: packet pulled from queue +0ms
  mqttjs:client _handlePacket :: emitting packetreceive +1ms
  mqttjs:client _handleConnack +0ms
  mqttjs:client _setupKeepaliveManager :: keepalive 1 (seconds) +0ms
  mqttjs:client KeepaliveManager: set keepalive to 1000ms +0ms
  mqttjs:client connect :: sending queued packets +1ms
  mqttjs:client deliver :: entry undefined +0ms
  mqttjs:client _resubscribe +0ms
  mqttjs:client _sendPing :: sending pingreq +1s
  mqttjs:client _sendPacket :: (mqttjs_4713df34) ::  start +0ms
  mqttjs:client _writePacket :: packet: { cmd: 'pingreq' } +1ms
  mqttjs:client _writePacket :: emitting `packetsend` +0ms
  mqttjs:client _writePacket :: writing to stream +1ms
  mqttjs:client _writePacket :: writeToStream result true +0ms
  mqttjs:client _writePacket :: invoking cb +0ms
  mqttjs:client noop :: undefined +0ms
  mqttjs:client writable stream :: parsing buffer +2ms
  mqttjs:client parser :: on packet push to packets array. +0ms
  mqttjs:client work :: getting next packet in queue +0ms
  mqttjs:client work :: packet pulled from queue +0ms
  mqttjs:client _handlePacket :: emitting packetreceive +0ms
  mqttjs:client _handlePacket :: received pingresp +0ms
Error: Keepalive timeout
    at MqttClient.onKeepaliveTimeout (C:\work\labs\node\mqtt\node_modules\mqtt\build\lib\client.js:1039:28)
    at Timeout._onTimeout (C:\work\labs\node\mqtt\node_modules\mqtt\build\lib\KeepaliveManager.js:67:29)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)
  mqttjs:client end :: (mqttjs_4713df34) +506ms
  mqttjs:client end :: cb? false +0ms
  mqttjs:client _clearReconnect : clearing reconnect timer +1ms
  mqttjs:client end :: (mqttjs_4713df34) :: immediately calling finish +0ms
  mqttjs:client end :: (mqttjs_4713df34) :: finish :: calling _cleanUp with force false +0ms
  mqttjs:client _cleanUp :: done callback provided for on stream close +0ms
  mqttjs:client _cleanUp :: forced? false +0ms
  mqttjs:client _cleanUp :: (mqttjs_4713df34) :: call _sendPacket with disconnect packet +0ms
  mqttjs:client _sendPacket :: (mqttjs_4713df34) ::  start +1ms
  mqttjs:client _writePacket :: packet: { cmd: 'disconnect' } +0ms
  mqttjs:client _writePacket :: emitting `packetsend` +0ms
  mqttjs:client _writePacket :: writing to stream +0ms
  mqttjs:client _writePacket :: writeToStream result true +0ms
  mqttjs:client _writePacket :: invoking cb +0ms
  mqttjs:client _cleanUp :: (mqttjs_4713df34) :: destroying stream +0ms
  mqttjs:client _destroyKeepaliveManager :: destroying keepalive manager +0ms
  mqttjs:client onKeepaliveTimeout :: calling _cleanUp with force true +0ms
  mqttjs:client _cleanUp :: forced? true +0ms
  mqttjs:client _cleanUp :: (mqttjs_4713df34) :: destroying stream +0ms
  mqttjs:client (mqttjs_4713df34)stream :: on close +1ms
  mqttjs:client _flushVolatile :: deleting volatile messages from the queue and setting their callbacks as error function +0ms
  mqttjs:client stream: emit close to MqttClient +1ms
  mqttjs:client close :: connected set to `false` +0ms
  mqttjs:client close :: clearing connackTimer +0ms
  mqttjs:client close :: calling _setupReconnect +0ms
  mqttjs:client _setupReconnect :: doing nothing... +0ms
  mqttjs:client end :: finish :: calling process.nextTick on closeStores +0ms
  mqttjs:client end :: closeStores: closing incoming and outgoing stores +0ms
  mqttjs:client end :: closeStores: emitting end +0ms
  mqttjs:client end :: closeStores: invoking callback with args +0ms
  mqttjs:client noop :: undefined +0ms
```

### PR DESCRIPTION
**Pull request by k-keisuke**
Fixes #1903

### CODE REVIEW

File: src/lib/client.ts [hunk 1]
```diff
@@ -2119,6 +2119,18 @@ export default class MqttClient extends TypedEventEmitter<MqttClientEventCallbac
...
+	 * Reschedule the ping interval regardless of reschedulePings value
+	 */
+	public reschedulePingForcely() {
+		if (
+			this.keepaliveManager &&
+			this.options.keepalive
+		) {
+			this._reschedulePing()
+		}
+	}
```
Comment (robertsLando) [inline src/lib/client.ts:2132]: What about adding a boolean to `reschedulePing` function like `force = false` inste4ad of creating a new method?

```ts
/**
	 * Reschedule the ping interval
	 */
	public reschedulePing(force = false) {
		if (
			this.keepaliveManager &&
			this.options.keepalive &&
			(force || this.options.reschedulePings)
		) {
			this._reschedulePing()
		}
	}

```

File: src/lib/handlers/index.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ const handle: PacketHandler = (client, packet, done) => {
 			break
 		case 'pingresp':
 			client.log('_handlePacket :: received pingresp')
-			client.reschedulePing()
+			client.reschedulePingForcely()
```
Comment (robertsLando) [inline src/lib/handlers/index.ts:59]:
```suggestion
			client.reschedulePing(true)
```

**Review by robertsLando — COMMENTED** [review]
Please add also a test to cover this and fix lint issues: `npm run lint-fix`

Test info: https://github.com/mqttjs/MQTT.js/blob/main/DEVELOPMENT.md#tests

**Comment by k-keisuke** [comment]
Thanks for reviewing. I fix these and add test.

**Review by robertsLando — APPROVED** [review]
LGTM

**Review by robertsLando — APPROVED** [review]
Tests are good now and code too, just fix lint issues please: `npm run lint-fix`

**Pull request merged**